Typecheck error when using 'type aliases' via imports (regression since 0.4.1)

I spotted this behaviour change in the type checker when trying to compile the `containers` library with juvix HEAD. Specifically there's a type error on this line https://github.com/anoma/juvix-containers/blob/851119ca6b4e94fab0ce7353c4b9d2930f53cc56/Data/Map.juvix#L36

```
/containers/Data/Map.juvix:36:20-25: error:
The expression Set.empty {_} has type:
  AVLTree (Binding _ _)
but is expected to have type:
  Set (Binding _ _)
make: *** [build/Map] Error 1
```

This behaviour change was introduced by (analysis using git bisect):
* https://github.com/anoma/juvix/pull/2219

For a more minimal reproduction, consider the following two modules:

Bug.juvix
```
module Bug;

type A := a : A;

B : Type;
B := A;

type C := mkC : B -> C;
```

Import.juvix
```
module Import;

import Bug open;

c : C := mkC a;
```

Run `juvix typecheck Import.juvix`

```
/Import.juvix:5:14-15: error:
The expression a has type:
  A
but is expected to have type:
  B
```

Note that the following type checks fine, so the import is necessary to trigger the bug:

```
module NoBug;

type A := a : A;

B : Type;
B := A;

type C := mkC : B -> C;

c : C := mkC a;
```

Import.juvix typechecks fine with Juvix v0.4.1.

--- a/src/Juvix/Compiler/Internal/Translation/FromInternal/Analysis/TypeChecking/Checker.hs
+++ b/src/Juvix/Compiler/Internal/Translation/FromInternal/Analysis/TypeChecking/Checker.hs
@@ -48,11 +48,16 @@ checkModuleNoCache ::
   Sem r Module
 checkModuleNoCache (ModuleIndex Module {..}) = do
   _moduleBody' <-
-    (evalState (mempty :: NegativeTypeParameters) . checkModuleBody) _moduleBody
+    evalState (mempty :: NegativeTypeParameters)
+      . checkModuleBody
+      $ _moduleBody
+  _moduleExamples <- mapM checkExample _moduleExamples
   return
     Module
       { _moduleBody = _moduleBody',
-        ..
+        _moduleName,
+        _moduleExamples,
+        _modulePragmas
       }
 
 checkModuleBody ::
@@ -60,8 +65,8 @@ checkModuleBody ::
   ModuleBody ->
   Sem r ModuleBody
 checkModuleBody ModuleBody {..} = do
-  _moduleStatements' <- mapM checkStatement _moduleStatements
   _moduleImports' <- mapM checkImport _moduleImports
+  _moduleStatements' <- mapM checkStatement _moduleStatements
   return
     ModuleBody
       { _moduleStatements = _moduleStatements',

--- a/src/Juvix/Compiler/Internal/Translation/FromInternal/Analysis/TypeChecking/Data/Inference.hs
+++ b/src/Juvix/Compiler/Internal/Translation/FromInternal/Analysis/TypeChecking/Data/Inference.hs
@@ -422,7 +422,7 @@ addIdens idens = do
 functionDefEval :: forall r'. Member (State FunctionsTable) r' => FunctionDef -> Sem r' (Maybe Expression)
 functionDefEval = runFail . goTop
   where
-    goTop :: forall r. (Members '[Fail, State FunctionsTable] r) => FunctionDef -> Sem r Expression
+    goTop :: forall r. Members '[Fail, State FunctionsTable] r => FunctionDef -> Sem r Expression
     goTop f =
       case f ^. funDefClauses of
         c :| [] -> goClause c
@@ -458,7 +458,7 @@ functionDefEval = runFail . goTop
             go :: [(PatternArg, Expression)] -> Sem r Expression
             go = \case
               [] -> return (c ^. clauseBody)
-              ((p, ty) : ps)
+              (p, ty) : ps
                 | Implicit <- p ^. patternArgIsImplicit -> fail
                 | otherwise -> go ps >>= goPattern (p ^. patternArgPattern, ty)
 

--- a/test/Typecheck/Positive.hs
+++ b/test/Typecheck/Positive.hs
@@ -236,7 +236,11 @@ tests =
     posTest
       "Nested pattern match with type variables"
       $(mkRelDir ".")
-      $(mkRelFile "NestedPatterns.juvix")
+      $(mkRelFile "NestedPatterns.juvix"),
+    posTest
+      "issue2248: Import type alias"
+      $(mkRelDir "issue2248")
+      $(mkRelFile "Main.juvix")
   ]
     <> [ compilationTest t | t <- Compilation.tests
        ]

--- a/tests/positive/issue2248/Bug.juvix
+++ b/tests/positive/issue2248/Bug.juvix
@@ -1,0 +1,8 @@
+module Bug;
+
+type A := mkA : A;
+
+aliasA : Type;
+aliasA := A;
+
+type C := mkC : aliasA -> C;

--- a/tests/positive/issue2248/Main.juvix
+++ b/tests/positive/issue2248/Main.juvix
@@ -1,0 +1,5 @@
+module Main;
+
+import Bug open;
+
+c : C := mkC mkA;

--- a/tests/positive/issue2248/juvix.yaml
+++ b/tests/positive/issue2248/juvix.yaml
@@ -1,0 +1,4 @@
+dependencies:
+- .juvix-build/stdlib/
+name: issue2248
+version: 0.0.0

